src/element/collision.ts

```diff
@@ -48,10 +45,26 @@ export const hitTest = (
 ): boolean => {
   // How many pixels off the shape boundary we still consider a hit
   const threshold = 10 / appState.zoom;
+
+  if (appState.selectedElementIds[element.id]) {
+    const [x1, y1, x2, y2] = getElementAbsoluteCoords(element);
+    const cx = (x1 + x2) / 2;
+    const cy = (y1 + y2) / 2;
+    // reverse rotate the pointer
+    [x, y] = rotate(x, y, cx, cy, -element.angle);
+
+    const doCoordinatesHitElementBoundingBox =
+      x > x1 - threshold &&
+      x < x2 + threshold &&
+      y > y1 - threshold &&
+      y < y2 + threshold;
+    return doCoordinatesHitElementBoundingBox;
+  }
```

Comment:
Maybe pull this out into a function to keep `hitTest` readable?

Reply:
I'll see what I manage to do to improve `hitTest` readability :)

Reply:
I've updated the PR with some refactors to `hitTest`.